### PR TITLE
Add pipeline for publishing cadl-ranch test coverage

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Replaced some relative paths with fully qualified ones. Updated to latest version of Node.

### CODE REVIEW

File: eng/pipelines/cadl-ranch-typespec.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - main
+  paths:
+    include:
+      - packages/typespec-go/test/cadlranch
+pr: none
+
+steps:
+  - template: ../steps/set-env.yaml
+    parameters:
+      NodeVersion: "18.x"
```
Comment (jhendrixMSFT) [inline eng/pipelines/cadl-ranch-typespec.yml:13]: @benbp do we need to set these params since they're already defined in `set-env.yaml`?

**Comment by jhendrixMSFT** [comment]
@benbp could you please help set up the internal pipeline for this? It's based on the [Python version](https://github.com/Azure/autorest.python/blob/getAllOperations/eng/pipelines/internal-ci.yml). The gist is that whenever we update the cadl-ranch tests in `main` we want this to trigger. It will also need some secrets wired in.

**Review by RickWinter — APPROVED** [review]

File: eng/pipelines/cadl-ranch-typespec.yml (hunk 1 above)
Comment (benbp) [reply eng/pipelines/cadl-ranch-typespec.yml:13]: If you're not trying to override the default then no.

**Comment by benbp** [comment]
@jhendrixMSFT https://dev.azure.com/azure-sdk/internal/_build/index?definitionId=6688&_a=completed

**Comment by jhendrixMSFT** [comment]
Thanks @benbp !

**Pull request merged**